readthedocs/proxito/views/hosting.py

```diff
@@ -0,0 +1,103 @@
+"""Views for hosting features."""
+
+import structlog
+from django.conf import settings
+from django.http import JsonResponse
+from django.views import View
+
+from readthedocs.builds.constants import EXTERNAL
+from readthedocs.core.mixins import CDNCacheControlMixin
+from readthedocs.core.unresolver import unresolver
+
+log = structlog.get_logger(__name__)  # noqa
+
+
+class ReadTheDocsConfigJson(CDNCacheControlMixin, View):
+    def get(self, request):
+
+        unresolved_domain = request.unresolved_domain
+        project = unresolved_domain.project
+
+        # TODO: why the UnresolvedURL object is not injected in the `request` by the middleware.
+        # Is is fine to calculate it here?
+        unresolved_url = unresolver.unresolve_url(request.headers.get("Referer"))
+        version = unresolved_url.version
```

Comment:
@stsewd seems like one for you. 

@humitos I see we're using the referer here. Is this because we don't want to pass in the domain explicitly as a GET arg? I think we likely want to actually use the `request.host` here, instead of a header that the browser is setting.

Reply:
> @humitos I see we're using the referer here. Is this because we don't want to pass in the domain explicitly as a GET arg? I think we likely want to actually use the request.host here, instead of a header that the browser is setting.

Not strong preference. We can use something different here. I used the `Referer` because it was what I had at hand and easy to get.

A `url=` GET attribute probably seems fine as well, since it's what we are using in other API endpoints (analytics and embed api, for example). Referer also makes testing harder since "copying and pasting a URL doesn't work" --I've ended up hardcoding the URL in the code because of this 😅 

I think we should combine both, `url=` GET attribute with a validation with `request.host` on that URL.